src/careamics/lightning/lightning_module.py

```diff
@@ -326,31 +344,49 @@ def training_step(self, batch: Tensor, batch_idx: Any) -> Any:
         # Compute loss
         loss = self.loss_func(out, target, self.loss_parameters)  # TODO ugly ?
 
-        self.log_dict(loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        # Logging
+        # TODO: implement a separate logging method?
+        self.log_dict(loss, on_step=True, on_epoch=True)
+        # self.log("lr", self, on_epoch=True)
```

Comment:
I donot see a check for NaN. That is important.  https://github.com/ashesh-0/Disentangle/blob/master/disentangle/nets/lvae.py#L856
If it is there at some other place, then all is good.

Reply:
Exactly. Specifically, it happens here: https://github.com/federico-carrara/careamics/blob/83583aa4803466fe79149f9ecb25eb4e89f961dc/src/careamics/losses/lvae/losses.py#L367